Mysql and BLOB column

## Setup

### Versions

- **Rust:** rustc 1.19.0
- **Diesel:** 0.15.2
- **Database:** Mysql
- **Operating System** OSX

### Feature Flags

- **diesel:** `mysql`
- **diesel_codegen:** `mysql`

## Problem Description
When using `infer_schema!` on a table that includes a BLOB column compilation does not complete successfully.

### What are you trying to accomplish?
Using a struct of the following definition:

``` rust
#[derive(Queryable)]
struct RSAKeyPairData {
    pub id: i64,
    pub private_key_pem: Vec<u8>,
    pub certificate_pem: Vec<u8>,
    pub published: bool,
    pub active: bool,
}
```

### What is the expected output?
Valid compilation.

### What is the actual output?
```
error[E0412]: cannot find type `Blob` in this scope
 --> src/schema.rs:5:1
  |
5 | infer_schema!("dotenv:DATABASE_URL");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
  |
  = help: there is an enum variant `diesel::mysql::MysqlType::Blob`, did you mean to use `diesel::mysql::MysqlType`?
  = note: this error originates in a macro outside of the current crate
```

### Are you seeing any additional errors?
No.

### Steps to reproduce

``` sql
CREATE TABLE rsakeypairs (
  id SERIAL PRIMARY KEY,
  private_key_pem BLOB NOT NULL,
  certificate_pem BLOB NOT NULL,
  published BOOLEAN NOT NULL DEFAULT 0,
  active BOOLEAN NOT NULL DEFAULT 0
)
```

Use `infer_schema!` to trigger.

n.b. `diesel print-schema` outputs:

```
table! {
    rsakeypairs (id) {
        id -> Bigint,
        private_key_pem -> Blob,
        certificate_pem -> Blob,
        published -> Bool,
        active -> Bool,
    }
}
```

Manually supplying the following `table!` definition as opposed to using `infer_schema!` removes the compilation fault and all appears to function correctly, I am not sure if `Binary` is correct here (though from docs it seems to be):

```
table! {
    rsakeypairs (id) {
        id -> Bigint,
        private_key_pem -> Binary,
        certificate_pem -> Binary,
        published -> Bool,
        active -> Bool,
    }
}
```

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.

--- a/diesel/src/types/mod.rs
+++ b/diesel/src/types/mod.rs
@@ -196,7 +196,7 @@ pub type VarChar = Text;
 
 /// The binary SQL type.
 ///
-/// On MySQL, it is also aliased by `Tinyblob`, `Mediumblob`, `Longblob`, `Bit` and `Varbinary`.
+/// On MySQL, it is also aliased by `Tinyblob`, `Blob`, `Mediumblob`, `Longblob`, `Bit` and `Varbinary`.
 ///
 /// ### [`ToSql`](/diesel/types/trait.ToSql.html) impls
 ///
@@ -212,6 +212,7 @@ pub type VarChar = Text;
 #[derive(Debug, Clone, Copy, Default)] pub struct Binary;
 
 #[doc(hidden)] pub type Tinyblob = Binary;
+#[doc(hidden)] pub type Blob = Binary;
 #[doc(hidden)] pub type Mediumblob = Binary;
 #[doc(hidden)] pub type Longblob = Binary;
 #[doc(hidden)] pub type Varbinary = Binary;

--- a/diesel_tests/tests/schema_inference.rs
+++ b/diesel_tests/tests/schema_inference.rs
@@ -208,3 +208,51 @@ mod postgres {
         assert_eq!(Ok(vec![inferred_ranges]), all_the_ranges::table.load(&conn));
     }
 }
+
+#[cfg(feature = "mysql")]
+mod mysql {
+    use diesel::*;
+    use schema::*;
+
+    #[derive(Insertable)]
+    #[table_name="all_the_blobs"]
+    struct InferredBlobs<'a> {
+        id: i32,
+        tiny: &'a [u8],
+        normal: &'a [u8],
+        medium: &'a [u8],
+        big: &'a [u8],
+    }
+
+    #[derive(Queryable, Debug, PartialEq)]
+    struct Blobs {
+        id: i32,
+        tiny: Vec<u8>,
+        normal: Vec<u8>,
+        medium: Vec<u8>,
+        big: Vec<u8>,
+    }
+
+    #[test]
+    fn blobs_are_correctly_inferred() {
+        let conn = connection();
+        let inferred_blobs = InferredBlobs {
+            id: 0,
+            tiny: &[0x01],
+            normal: &[0x02],
+            medium: &[0x03],
+            big: &[0x04],
+        };
+
+        let blobs = Blobs {
+            id: 0,
+            tiny: vec![0x01],
+            normal: vec![0x02],
+            medium: vec![0x03],
+            big: vec![0x04],
+        };
+
+        insert(&inferred_blobs).into(all_the_blobs::table).execute(&conn).unwrap();
+        assert_eq!(Ok(vec![blobs]), all_the_blobs::table.load(&conn));
+    }
+}

--- a/migrations/mysql/20170811104602_add_blob_types/down.sql
+++ b/migrations/mysql/20170811104602_add_blob_types/down.sql
@@ -1,0 +1,1 @@
+DROP TABLE all_the_blobs;

--- a/migrations/mysql/20170811104602_add_blob_types/up.sql
+++ b/migrations/mysql/20170811104602_add_blob_types/up.sql
@@ -1,0 +1,7 @@
+CREATE TABLE all_the_blobs (
+    id INTEGER PRIMARY KEY, -- Can't use a blob as a pk
+    tiny TINYBLOB NOT NULL,
+    normal BLOB NOT NULL,
+    medium MEDIUMBLOB NOT NULL,
+    big LONGBLOB NOT NULL
+)

